ros2interface/ros2interface/verb/show.py

```diff
@@ -47,8 +108,44 @@ def add_arguments(self, parser, cli_name):
 
     def main(self, *, args):
         try:
-            file_path = get_interface_path(args.type)
-        except LookupError as e:
+            self._expand_message(interface_identifier=args.type)
+        except (ValueError, LookupError) as e:
             return str(e)
+
+    def _expand_message(self, interface_identifier: str):
+        lines = self._get_interface_content(interface_identifier, indent_level=0)
+        line_idx = 0
+        while line_idx < len(lines):
+            line = lines[line_idx]
+            if line.nested_type:
+                new_lines = self._get_interface_content(
+                    line.nested_type,
+                    indent_level=line.indent_level+1,
```

Comment:
It seems like `indent_level` is passed to places that don't need knowledge of it. First, it's initialized in `_expand_message`, but `_expand_message()` doesn't actually use it. It's passed to `_get_interface_content()` here, but that passes it up to `TextLine` which is again returned to `_expand_message()` by way of the `str()` call when this prints the line. It seems like `_expand_message` is the only function that needs to know about the indent level.

What if this used a stack to track the indentation instead of passing that info up and down?

```python3
# List used as a stack, each item is list of lines in a message
interface_stack = [self._get_interface_content(interface_dentifier)]
while interface_stack:
    interface = interface_stack[-1]
    while interface:
        line = interface.pop()
        # Size of stack is `indent_level`
        print('\t' * len(interface_stack - 1) + line)
        if line.nested_type:
            # Found a message that needs handling, add it to top of stack
            interface_stack.append(self._get_interface_content(line.nested_type))
            # Pause handling this interface to handle deeper one
            break
    if not interface:
        # finished handling this interface, remove it from stack
        interface_stack.pop()
```

Reply:
Good idea!